backend/node/views.py

```diff
@@ -0,0 +1,113 @@
+from django.shortcuts import render
+from rest_framework.generics import GenericAPIView
+from .models import Node
+# Create your views here.
+from rest_framework.decorators import api_view, permission_classes
+
+from rest_framework import response, status
+from rest_framework.decorators import api_view
+from rest_framework.permissions import AllowAny, IsAuthenticated, IsAuthenticatedOrReadOnly
+#import 
+import requests
+from rest_framework import authentication
+
+import base64
+
+"""
+I thought we didnt need generic view here
+
+"""
+
+credentialForConnect = {"username" : "hello", "password" : "world"}  # credential to connect
+credentialForDelete = {"", "", "", ""}
+
+
+# TODO:   this permission class DOES NOT WORK. get request still goes to my customAuthentication class in nodeFunction.py
+@api_view(['GET', 'DELETE'])
```

Comment:
Should use POST instead. GET isn't typically used when creating new objects. POST can also allow for your idea of other nodes sending their hostname through the body. 
```suggestion
@api_view(['POST',])
```